extensions/objects/templates/image-editor.component.ts

```diff
@@ -0,0 +1,1155 @@
+        // handle transparent content. These GIFs have 0 or 1 Disposal
+        // method value.
+        // See https://www.w3.org/Graphics/GIF/spec-gif89a.txt
+        if (frameData[i].frameInfo.disposal > 1) {
+          // Frames that have transparent content may not render
+          // properly in the gifshot output. As a workaround, add a
+          // white background to individual frames before creating a
+          // GIF.
+          let ctx = sourceCanvas.getContext('2d');
+          ctx.globalCompositeOperation = 'destination-over';
+          ctx.fillStyle = '#FFF';
+          ctx.fillRect(0, 0, sourceCanvas.width, sourceCanvas.height);
+          ctx.globalCompositeOperation = 'source-over';
+        }
+        let dataURL = sourceCanvas.toDataURL('image/png');
+        let updatedFrame = (
+          processFrameCallback ?
+          await processFrameCallback(dataURL) : dataURL);
+        frames.push(updatedFrame);
+      }
+      gifshot.createGIF({
+        gifWidth: width,
+        gifHeight: height,
+        images: frames
+      }, successCallback);
+    });
+  }
+
+  saveImageToLocalStorage(
+      dimensions: Dimensions,
+      resampledFile: Blob,
+      imageType: string): void {
+    const filename = this.imageUploadHelperService.generateImageFilename(
+      dimensions.height, dimensions.width, imageType);
+    const reader = new FileReader();
+    reader.onload = () => {
+      const imageData = reader.result as string;
+      // Check point 2 in the note before imports and after fileoverview.
+      this.imgData = imageData;
+      this.imageLocalStorageService.saveImage(filename, imageData);
+      this.setSavedImageFilename(filename, true);
+      const dimensions = (
+        this.imagePreloaderService.getDimensionsOfImage(filename));
+      this.imageContainerStyle = {
+        height: dimensions.height + 'px',
+        width: dimensions.width + 'px'
+      };
+    };
+    reader.readAsDataURL(resampledFile);
+  }
+
+  saveImage(
+      dimensions: Dimensions,
+      resampledFile: Blob,
+      imageType: string): void {
+    if (
+      this.contextService.getImageSaveDestination() ===
+      AppConstants.IMAGE_SAVE_DESTINATION_LOCAL_STORAGE) {
+      this.saveImageToLocalStorage(dimensions, resampledFile, imageType);
+    } else {
+      this.postImageToServer(dimensions, resampledFile, imageType);
+    }
+  }
+
+  postImageToServer(
+      dimensions: Dimensions,
+      resampledFile: Blob,
+      imageType: string = 'png'): void {
+    let form = new FormData();
+    form.append('image', resampledFile);
+    form.append('payload', JSON.stringify({
+      filename: this.imageUploadHelperService.generateImageFilename(
+        dimensions.height, dimensions.width, imageType)
+    }));
+    const imageUploadUrlTemplate = '/createhandler/imageupload/' +
+      '<entity_type>/<entity_id>';
+    this.csrfTokenService.getTokenAsync().then((token) => {
+      form.append('csrf_token', token);
+      $.ajax({
```

Comment:
Can we use something else than jQuery?

Reply:
I think this requires more refactoring and is out of scope of this PR (this PR just moves existing logic to a new file i.e. from filepath editor to image editor).